packages/storybook/src/stories/grid-pagination.stories.tsx

```diff
@@ -199,10 +210,104 @@ export function AutoPagination() {
           columns={data.columns}
           options={{
             pagination: true,
-            paginationAutoPageSize: true,
+            autoPageSize: true,
           }}
         />
       </div>
     </React.Fragment>
   );
 }
+
+function loadServerRows(params: PageChangedParams): Promise<GridData> {
+  return new Promise<GridData>((resolve) => {
+    getData(params.pageSize, 10).then((data) => {
+      setTimeout(() => {
+        const minId = (params.page - 1) * params.pageSize;
+        data.rows.forEach((row) => {
+          row.id = (Number(row.id) + minId).toString();
+        });
+        resolve(data);
+      }, 500);
+    });
+  });
+}
+
+export function ServerPaginationWithApi() {
+  const apiRef: ApiRef = useApiRef();
+  const data = useData(100, 10);
+  const [rows, setRows] = useState<RowsProp>([]);
+  const [isLoading, setLoading] = useState<boolean>(false);
+
+  React.useEffect(() => {
+    let unsubscribe;
+    if (apiRef && apiRef.current) {
```

Comment:
This is already guaranteed by React

```suggestion
```

@eps1lon Any idea on how to best handle the TypeScript implication, simply a `// @ts-ignore`?

Reply:
I wouldn't guard against it. Or at least come up with a strategy when you encounter an unexpected state. These should be resolve to actual Elements. If there's a reason why then you should document it and, even better, add a test. Because that would be unusual.

Reply:
@eps1lon so you would do something like this instead?

```ts
if (!apiRef.current) {
  throw new Error('missing ref binding');
}
```

Reply:
Yeah but a better error message (what is a ref binding? which ref? where?). If this is not something that passes through user code then explicitly mention that this is a bug in Material-UI and and that they should open an issue. React has those as well and I'd rather not loose an issue report because someone gave up after debugging for hours when they should've just opened an issue.

Reply:
I'll remove the if check, as apiRef is already initialized at this stage.